packages/crypto/src/hashing.ts

```diff
@@ -47,18 +47,24 @@ async function hashTransaction(tx: TxBody, encoding: 'bytes', includeSign?: bool
 async function hashTransaction(tx: TxBody, encoding = 'base64', includeSign = true): Promise<Buffer | string> {
     // check amount format
     tx.amount = '' + tx.amount;
+    const amount = tx.amount.replace(/[^0-9]/g,'');
```

Comment:
how about removing line 49 and put
```suggestion
const amount = tx.amount.toString().replace(/[^0-9]/g,'');
```
instead. (to decrease side effects on tx)